examples/gitserver/README.md

```diff
@@ -1,16 +1,220 @@
+It can clone repositories from remote systems on startup as well as automatically
+launch builds in an OpenShift project. It can automatically initialize and receive
+Git directories on push.
+
+Hooks can be customized to perform actions on push such as invoking `oc new-app` on a
+repository's source.
+
+The Dockerfile built by this example is published as openshift/origin-gitserver
+
+Quick Start
+-----------
+
+Prerequisites:
+
+* You have an OpenShift v3 server running
+* You are logged in and have access to a project
+* You have the `gitserver.yaml` in this directory
+* You can create externally accessible routes on your server
+
+**NOTE:** It is strongly recommended to use a secure route for your server. Otherwise
+all code and credentials are passed in plain text between your client and the server.
+
+### Deploy the Git Server
+
+1. Create the Git Server
+
+    ```sh
+    $ oc create -f gitserver.yaml
+    ```
+
+2. Grant `view` access to the `gitserver` service account
+
+    ```sh
+    $ oc policy add-role-touser view -z gitserver
+    ```
+
+3. Create a route to the Git Server
+
+    ```sh
+    $ oc expose service/gitserver
+    ```
+
+    This should assign a default hostname to your Git Server using the name of your
+    current project. To see the hostname that was assigned to your Git Server, execute:
+
+    ```sh
+    $ oc get route gitserver
+    ```
+
+    The hostname should be something like: `gitserver-myproject.infra.openshift.com`
+    where `myproject` is the name of your project, and `.infra.openshift.com` is the default
+    suffix for routes in your cluster. In the instructions that follow please substitute
+    `gitserver-myproject.infra.openshift.com` with the host name given to you by your 
+    cluster.
+
+4. Update the PUBLIC_URL of your Git Server with the one provided by the route
+
+    ```sh
+    $ oc env dc/gitserver PUBLIC_URL=http://gitserver-myproject.infra.openshift.com
+    ```
+
+    This will result in a new deployment of the Git Server with the updated environment 
+    variable.
+
+
+### Push code to the Git Server
+
+Code repositories can be initialized on the Git Server by either doing a 'git push' or
+a 'git clone' of the repository.
+
+1. Setup your credentials
+
+   By default the Git Server will allow users or service accounts that can create pods in
+   the project namespace to create and push code to the Git Server. The easiest way to 
+   provide credentials to the Git Server is by using a custom credential helper that will 
+   send your OpenShift token by default to the server.
+
+   ```sh
+   $ git config --global credential.http://gitserver-myproject.infra.openshift.com.helper \
+         '!f() { echo "username=$(oc whoami)"; echo "password=$(oc whoami -t)"; }; f'
+   ```
+
+   **NOTE:** the config key is `credential.[your server URL here].helper`
+
+2. Clone a new repository
+   
+   ```
+   $ git clone http://gitserver-myproject.infra.openshift.com/ruby-hello-world.git
```

Comment:
No reason... it's just one possible flow. If pushing first is more common, I'll use that.